pheweb/serve/data_access/db.py

```diff
@@ -1052,9 +1159,80 @@ def get_lofs(self, gene):
+          self.user = getattr(auth_module, 'mysql')['user']
+          self.password = getattr(auth_module, 'mysql')['password']
+          self.host = getattr(auth_module, 'mysql')['host']
+          self.db = getattr(auth_module, 'mysql')['db']
+          self.release = getattr(auth_module, 'mysql')['release']
+     def get_connection(self):
+          return pymysql.connect(host=self.host, user=self.user, password=self.password, db=self.db)
+     def get_max_region(self, phenocode, chr, start, end):
+          conn = self.get_connection()
+          with conn.cursor(pymysql.cursors.DictCursor) as cursori:
+               sql = "SELECT min(start) as start, max(end) AS end FROM finemapped_regions WHERE rel=%s AND phenocode=%s AND chr=%s AND start <= %s AND end >= %s"
+               cursori.execute(sql, [self.release, phenocode, chr, end, start])
+               result = cursori.fetchone()
+               cursori.close()
+          conn.close()
+          return result
+     def get_regions(self, variant: Variant):
+          conn = self.get_connection()
+          with conn.cursor(pymysql.cursors.DictCursor) as cursori:
+               sql = "SELECT type, phenocode, chr, start, end, path FROM finemapped_regions WHERE rel=%s AND chr=%s AND start <= %s AND end >= %s"
+               cursori.execute(sql, [self.release, variant.chr, variant.pos, variant.pos])
+               result = cursori.fetchall()
+               for res in result:
+                    res['path'] = self.base_paths[res['type']] + '/' + res['path']
+          conn.close()
+          return result
+     def get_regions_for_pheno(self, type, phenocode, chr, start, end, get_most_probable_finemap_n=True):
+          conn = self.get_connection()
+          with conn.cursor(pymysql.cursors.DictCursor) as cursori:
+               if type == 'all':
+                    sql = "SELECT type, chr, start, end, n_signals, n_signals_prob, variants, path FROM finemapped_regions WHERE rel=%s AND phenocode=%s AND chr=%s AND start <= %s AND end >= %s"
+                    cursori.execute(sql, [self.release, phenocode, chr, end, start])
+               elif type == 'conditional':
+                    sql = "SELECT type, chr, start, end, n_signals, variants, path FROM finemapped_regions WHERE rel=%s AND type=%s AND phenocode=%s AND chr=%s AND start <= %s AND end >= %s"
+                    cursori.execute(sql, [self.release, 'conditional', phenocode, chr, end, start])
+               elif type == 'finemapping':
+                    sql = "SELECT type, chr, start, end, n_signals, n_signals_prob, path FROM finemapped_regions WHERE rel=%s AND (type=%s OR type=%s) AND phenocode=%s AND chr=%s AND start <= %s AND end >= %s"
+                    cursori.execute(sql, [self.release, 'susie', 'finemap', phenocode, chr, end, start])
+               else:
+                    raise ValueError('unsupported type "' + type + '"')
+               result = cursori.fetchall()
```

Comment:
I tried to look for pymsql docs for return type of fetchall but it was noe specified. What are the return types?

Reply:
DictCursor gives dict